### ISSUE TITLE
is hard Gumbel used in paper?

### ISSUE THREAD

**Issue by ysymyth** [cambridgeltl/ECNMT#3]
I see args.hard is default False, and when I tried with it it doesn't really work well?

**Comment by yaoyiran** [comment]
We only reported the results based on the soft version soft-max.

**Comment by yaoyiran** [comment]
To make the EC pretraining more stable, you could try the following: (1) set the learning rate as 1e-4 rather than the default 1e-3; (2) set a minimum message length in models.py, e.g. replacing line 26 "if idx_ == 0:" with , say, "if idx_ <= 5:", "if idx_ <= 10:" or "if idx_ >= 0:".
